### ISSUE TITLE
Introduce a loop detection 

### ISSUE THREAD

**Issue by mcweba** [swisspush/gateleen#136]
## Problem
The following routing rule forwards the incoming request to itself:
```
"/playground/looping/(.*)": {
	"description": "looping Test",
	"metricName": "looperRule",
	"path": "/playground/looping/$1"
}
```
This routing rule creates a loop which will repeat until the server is dead. The result is a _java.lang.StackOverflowError_ exception. After this exception, the server is not able to handle any more requests and have to be restarted.

## Solution
The idea is to add a header containing the count of routings to the request.

The header could be called _**x-hops**_ and will be checked against a configurable limit before re-routing in the _Router_. When the limit is not yet reached, the _**x-hops**_ header value will be increased by 1.

When the limit is reached, the request will be terminated with a http status code 500 (internal server error) with a corresponding message like 'loop detected'.

**Comment by lbovet** [comment]
Yes, makes sense. A short header like "x-hops".
This would be a thing common to all Forwarders.

**Comment by mcweba** [comment]
"x-hops" sounds good :-). A first test showed that this feature can be implemented in the _Router_ class. Therefore it automatically works for all Forwarders

**Comment by mcweba** [comment]
Fixed in release [v1.0.48](https://github.com/swisspush/gateleen/releases/tag/v1.0.48)
